File: domain/src/test/java/com/arturogutierrez/openticator/domain/category/CategoryFactoryTest.java
package com.arturogutierrez.openticator.domain.category;

import com.arturogutierrez.openticator.domain.category.model.Category;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.notNullValue;
import static org.hamcrest.MatcherAssert.assertThat;

public class CategoryFactoryTest {

  private CategoryFactory categoryFactory;

  @Before
  public void setUp() {
    categoryFactory = new CategoryFactory();
  }

  @Test
  public void testCreatingCategory() {
    Category category = categoryFactory.createCategory("name");

    assertThat(category.getCategoryId(), is(notNullValue()));
    assertThat(category.getName(), is("name"));
  }
}

File: domain/src/test/kotlin/com/arturogutierrez/openticator/domain/category/CategoryFactoryTest.kt
package com.arturogutierrez.openticator.domain.category

import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.CoreMatchers.notNullValue
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Before
import org.junit.Test

class CategoryFactoryTest {

  private lateinit var categoryFactory: CategoryFactory

  @Before
  fun setUp() {
    categoryFactory = CategoryFactory()
  }

  @Test
  fun testCreatingCategory() {
    val category = categoryFactory.createCategory("name")

    assertThat(category.categoryId, `is`(notNullValue()))
    assertThat(category.name, `is`("name"))
  }
}
